src/phoenix/server/app.py

```diff
@@ -302,8 +302,48 @@ async def api_docs(request: Request) -> Response:
     return get_swagger_ui_html(openapi_url="/schema", title="arize-phoenix API")
 
 
-def create_app(
+class SessionFactory:
+    def __init__(
+        self,
+        session_factory: Callable[[], AsyncContextManager[AsyncSession]],
+        dialect: str,
+        engine: Optional[AsyncEngine] = None,
+    ):
+        self.session_factory = session_factory
+        self.dialect = dialect
+        self.engine = engine
+
+    def __call__(self) -> AsyncContextManager[AsyncSession]:
+        return self.session_factory()
+
+
+def create_engine_and_run_migrations(
     database_url: str,
+) -> SessionFactory:
+    try:
+        engine = create_engine(database_url)
+    except PhoenixMigrationError as e:
+        msg = (
+            "\n\n⚠️⚠️ Phoenix failed to migrate the database to the latest version. ⚠️⚠️\n\n"
+            "The database may be in a dirty state. To resolve this, the Alembic CLI can be used\n"
+            "from the `src/phoenix/db` directory inside the Phoenix project root. From here,\n"
+            "revert any partial migrations and run `alembic stamp` to reset the migration state,\n"
+            "then try starting Phoenix again.\n\n"
+            "If issues persist, please reach out for support in the Arize community Slack:\n"
+            "https://arize-ai.slack.com\n\n"
+            "You can also refer to the Alembic documentation for more information:\n"
+            "https://alembic.sqlalchemy.org/en/latest/tutorial.html\n\n"
+            ""
+        )
+        raise PhoenixMigrationError(msg) from e
+    factory = SessionFactory(
+        session_factory=_db(engine), dialect=engine.dialect.name, engine=engine
+    )
+    return factory
+
+
+def create_app(
+    db: SessionFactory,
```

Comment:
This doesn't seem like the right name for this variable.

Reply:
the interface to this variable is the same as it is before, it's basically a sessionmaker and I think `db` is a pretty common shorthand for this sort of interface. What other names did you have in mind?

Reply:
Looking again, I think it's fine.